sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -555,6 +555,17 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val MAX_NESTED_VIEW_DEPTH =
+    buildConf("spark.sql.view.maxNestedViewDepth")
+      .internal()
+      .doc("The maximum level of view references allowed in a nested view. A nested view may " +
+        "reference other nested views, the dependencies are organized in a directed acyclic " +
+        "graph (DAG). However the DAG depth may grow really large and cause unexpected " +
+        "behaviors, so we put some limit on this: when the nested level of a view exceeds this " +
+        "value during view resolution, we terminate the resolution to avoid further errors.")
+      .intConf
+      .createWithDefault(100)
```

Comment:
Add `.checkValue(...)`

What is the behaviors of edge scenarios? Does that mean we can disable view support using this flag?